src/DetailPage.js

```diff
@@ -1,11 +1,44 @@
-import { useEffect, useState } from 'react';
-import { useRouteMatch } from 'react-router-dom';
+import { useEffect, useState, } from 'react';
+import { useRouteMatch, useParams, useHistory } from 'react-router-dom';
+import { updateGame } from './services/fetch-utils';
 import { getGameById } from './services/fetch-utils';
 
 export default function DetailPage() {
   const [game, setGame] = useState({});
+  const match = useRouteMatch();
+  const { push } = useHistory();
+  const [title, setTitle] = useState('');
+  const [genre, setGenre] = useState('');
+  const [designer, setDesigner] = useState('');
+  const [description, setDescription] = useState('');
+  const [minPlayers, setMinPlayers] = useState(0);
+  const [maxPlayers, setMaxPlayers] = useState(0);
+  const { id } = useParams();
+  useEffect(() => {
+    async function fetch() {
+      const gameResponse = await getGameById(match.params.id);
+      setGame(gameResponse);
+    }
+    fetch();
+  }, [match]);
 
-  // on mount, fetch and set in state the correct board game for this id (the id can be found in match.params using the correct react-router hook)
+  async function handleSubmit(e) {
+    e.preventDefault();
+
+    await updateGame({
+      title,
+      genre,
+      designer,
+      description,
+      min_players: minPlayers,
+      max_players: maxPlayers,
+      params: id
```

Comment:
I suspect what you are wanting to do here is pass in the updated game object AND the id to match it for an update. The way it's currently written I get this error:
![image](https://user-images.githubusercontent.com/73324535/168684343-b454d3dd-79a9-4911-8936-abf1ff7ab3fc.png)
 so you will probably want to do something like `await updateGame(game, id)` and then your updateGame function would take in two arguments: the game object and the id so you can do a match on id